dashboard/test/lib/homepage/homepage_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+
+class HomepageTest < ActiveSupport::TestCase
+  setup_all do
+    @request = ActionDispatch::Request.new({})
+  end
+
+  test 'banner with showInternationally false is seen only by users in the US and with unknown country' do
+    Homepage.class_variable_set(:@@json_path, File.join("#{dashboard_dir}/test/lib/homepage", 'homepage_test_in_US_or_unknown_location_banner.json'))
+
+    @request.stubs(:country).returns("US")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-in-US-or-unknown-location", banner[:id])
+
+    @request.stubs(:country).returns(nil)
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-in-US-or-unknown-location", banner[:id])
+
+    @request.stubs(:country).returns("AF")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    refute banner
+  end
+
+  test 'banner with showInternationally true is seen only by users we know are outside the US' do
+    Homepage.class_variable_set(:@@json_path, File.join("#{dashboard_dir}/test/lib/homepage", 'homepage_test_outside_US_banner.json'))
+
+    @request.stubs(:country).returns("US")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    refute banner
+
+    @request.stubs(:country).returns(nil)
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    refute banner
+
+    @request.stubs(:country).returns("AF")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-outside-US", banner[:id])
+  end
+
+  test 'banner without showInternationally flag is seen by everyone' do
+    Homepage.class_variable_set(:@@json_path, File.join("#{dashboard_dir}/test/lib/homepage", 'homepage_test_everywhere_banner.json'))
+
+    @request.stubs(:country).returns("US")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-everywhere", banner[:id])
+
+    @request.stubs(:country).returns(nil)
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-everywhere", banner[:id])
+
+    @request.stubs(:country).returns("AF")
+    banner = Homepage.get_announcement_for_page("homepage", @request)
+    assert_equal("show-everywhere", banner[:id])
+  end
+end
```

Comment:
these tests are awesome, thanks for adding these!

I feel bad even mentioning this, but as part of the effort to separate dashboard from pegasus, we are trying to avoid cross-project dependencies like this, and so in the future any tests on pegasus pages should go in pegasus/test. this is the ideal dependency graph:

<img width="1181" alt="Screenshot 2023-04-19 at 9 29 26 AM" src="https://user-images.githubusercontent.com/8001765/233140274-d4b373b3-46e9-4622-9ed2-9a0f272a1d74.png">

 this is my bad for not setting up linting rules preventing this kind of cross-project dependency, which I have been meaning to do for some time now. so for now I'm just going to ask that any additional tests for pegasus code be located in pegasus/test. 🙏

Reply:
Oh good call –– @kelbyhawn can I go ahead and update this? I hate shipping code that needs a refactor 😅

Reply:
@megcrenshaw yes, absolutely! How much time do you think this'll add before it can be merged? (Only asking due to the AI banner dependency 🙃)

Reply:
Give me like 15 minutes –– I have it moved with tests passing, but cleaning up a few things. Then it needs to go through drone ... so end of day your time

Reply:
Just pushed new stuff. @davidsbailey do you want to take another look? It's my first rodeo writing a pegasus unit test, but the test structure didn't change. Perhaps if there are requested refactors, I can tackle them in a separate PR